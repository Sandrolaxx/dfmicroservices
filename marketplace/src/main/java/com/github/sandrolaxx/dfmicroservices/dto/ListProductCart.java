package com.github.sandrolaxx.dfmicroservices.dto;

import java.util.List;

public class ListProductCart {
    
    List<ProductCart> listToUpdate;

    public List<ProductCart> getListToUpdate() {
        return listToUpdate;
    }

    public void setListToUpdate(List<ProductCart> listToUpdate) {
        this.listToUpdate = listToUpdate;
    }
    
}
