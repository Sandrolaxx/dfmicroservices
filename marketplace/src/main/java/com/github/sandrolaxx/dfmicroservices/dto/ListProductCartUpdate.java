package com.github.sandrolaxx.dfmicroservices.dto;

import java.util.List;

public class ListProductCartUpdate {
    
    List<ProductCartUpdate> listToUpdate;

    public List<ProductCartUpdate> getListToUpdate() {
        return listToUpdate;
    }

    public void setListToUpdate(List<ProductCartUpdate> listToUpdate) {
        this.listToUpdate = listToUpdate;
    }
    
}
