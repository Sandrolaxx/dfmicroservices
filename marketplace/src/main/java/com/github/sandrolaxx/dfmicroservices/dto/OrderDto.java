package com.github.sandrolaxx.dfmicroservices.dto;

import java.util.List;

import com.github.sandrolaxx.dfmicroservices.entities.enums.EnumOrderStatus;
import com.github.sandrolaxx.dfmicroservices.entities.enums.EnumPaymentType;

public class OrderDto {

    private Integer orderId;
    
    private List<ProductOrderDto> productsOrder;

    private EnumOrderStatus orderStatus;
    
    private EnumPaymentType payType;

    private String district;

    private String street;
    
    private Integer number;

    private Integer numberAp;

    private Double latitude;

    private Double longitude;

    private Double deliveryValue;
    
    private Double total;

    public Integer getOrderId() {
        return orderId;
    }

    public void setOrderId(Integer orderId) {
        this.orderId = orderId;
    }

    public List<ProductOrderDto> getProductsOrder() {
        return productsOrder;
    }

    public void setProductsOrder(List<ProductOrderDto> productsOrder) {
        this.productsOrder = productsOrder;
    }

    public EnumOrderStatus getOrderStatus() {
        return orderStatus;
    }

    public void setOrderStatus(EnumOrderStatus orderStatus) {
        this.orderStatus = orderStatus;
    }

    public EnumPaymentType getPayType() {
        return payType;
    }

    public void setPayType(EnumPaymentType payType) {
        this.payType = payType;
    }

    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public Double getDeliveryValue() {
        return deliveryValue;
    }

    public void setDeliveryValue(Double deliveryValue) {
        this.deliveryValue = deliveryValue;
    }

    public Double getTotal() {
        return total;
    }

    public void setTotal(Double total) {
        this.total = total;
    }

    public String getDistrict() {
        return district;
    }

    public void setDistrict(String district) {
        this.district = district;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public Integer getNumber() {
        return number;
    }

    public void setNumber(Integer number) {
        this.number = number;
    }

    public Integer getNumberAp() {
        return numberAp;
    }

    public void setNumberAp(Integer numberAp) {
        this.numberAp = numberAp;
    }
    
}
