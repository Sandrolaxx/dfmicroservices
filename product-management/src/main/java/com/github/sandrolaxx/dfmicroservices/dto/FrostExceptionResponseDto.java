package com.github.sandrolaxx.dfmicroservices.dto;

/**
 *
 * @author SRamos
 */
public class FrostExceptionResponseDto {

    private String errorDate;

    private String errorMessage;

    private String errorCode;

    private String httpCodeMessage;

    public String getErrorDate() {
        return this.errorDate;
    }

    public void setErrorDate(String errorDate) {
        this.errorDate = errorDate;
    }
    
    public String getErrorMessage() {
        return this.errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    public String getErrorCode() {
        return this.errorCode;
    }

    public void setErrorCode(String errorCode) {
        this.errorCode = errorCode;
    }

    public String getHttpCodeMessage() {
        return this.httpCodeMessage;
    }

    public void setHttpCodeMessage(String httpCodeMessage) {
        this.httpCodeMessage = httpCodeMessage;
    }

}
