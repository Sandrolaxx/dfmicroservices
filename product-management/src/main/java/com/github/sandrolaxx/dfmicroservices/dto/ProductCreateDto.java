package com.github.sandrolaxx.dfmicroservices.dto;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import javax.validation.constraints.Size;

public class ProductCreateDto {

  @NotNull
  @Size(min = 3, max = 40)
  public String name;

  @NotNull
  @Positive
  public Double price;

  @NotNull
  @Size(min = 5, max = 90)
  public String description;

  public String imageUri;

  @NotNull
  public Boolean active;

  public String getName() {
    return this.name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public Double getPrice() {
    return this.price;
  }

  public void setPrice(Double price) {
    this.price = price;
  }

  public String getDescription() {
    return this.description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getImageUri() {
    return this.imageUri;
  }

  public void setImageUri(String imageUri) {
    this.imageUri = imageUri;
  }

  public Boolean getActive() {
    return this.active;
  }

  public void setActive(Boolean active) {
    this.active = active;
  }

}
