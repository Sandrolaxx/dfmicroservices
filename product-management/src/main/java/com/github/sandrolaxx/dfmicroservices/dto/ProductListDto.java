package com.github.sandrolaxx.dfmicroservices.dto;

public class ProductListDto {

  public Long id;

  public String name;

  public Double price;

  public String description;

  public String imageUri;

  public Boolean active;

  public String createdAt;

  public String updatedAt;

  public Long getId() {
    return this.id;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public String getName() {
    return this.name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public Double getPrice() {
    return this.price;
  }

  public void setPrice(Double price) {
    this.price = price;
  }

  public String getDescription() {
    return this.description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getImageUri() {
    return this.imageUri;
  }

  public void setImageUri(String imageUri) {
    this.imageUri = imageUri;
  }

  public Boolean getActive() {
    return this.active;
  }

  public void setActive(Boolean active) {
    this.active = active;
  }

  public String getCreatedAt() {
    return this.createdAt;
  }

  public void setCreatedAt(String createdAt) {
    this.createdAt = createdAt;
  }

  public String getUpdatedAt() {
    return this.updatedAt;
  }

  public void setUpdatedAt(String updatedAt) {
    this.updatedAt = updatedAt;
  }

}
