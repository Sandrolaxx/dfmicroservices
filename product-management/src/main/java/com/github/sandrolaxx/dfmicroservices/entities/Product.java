package com.github.sandrolaxx.dfmicroservices.entities;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

@Entity
@Table(name = "DF_PRODUCT")
public class Product extends PanacheEntityBase {
  
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  public Integer id;//Panache já em tempo de execução coloca como private

  @Column(name = "NAME")
  public String name;
  
  @Column(name = "PRICE")
  public Double price; 
  
  @Column(name = "DESCRIPTION")
  public String description;
  
  @Column(name = "IMAGE_URI")
  public String imageUri;

  @Column(name = "ACTIVE")
  public Boolean active;

  @CreationTimestamp
  @Temporal(TemporalType.TIMESTAMP)
  @Column(name = "CREATED_AT")
  public Date createdAt;

  @UpdateTimestamp
  @Temporal(TemporalType.TIMESTAMP)
  @Column(name = "UPDATED_AT")
  public Date updatedAt;

}
