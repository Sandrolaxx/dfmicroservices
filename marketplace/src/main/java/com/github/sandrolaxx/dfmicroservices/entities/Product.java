package com.github.sandrolaxx.dfmicroservices.entities;

import java.sql.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

import com.github.sandrolaxx.dfmicroservices.dto.ProductDto;

import io.quarkus.hibernate.reactive.panache.PanacheEntityBase;

@Entity
@Table(name = "DF_PRODUCT")
public class Product extends PanacheEntityBase {

  @Id
  public Integer id;

  @Column(name = "NAME")
  public String name;
  
  @Column(name = "PRICE")
  public Double price;
  
  @Column(name = "DISCOUNT")
  public Double discount;

  @Column(name = "DESCRIPTION")
  public String description;
  
  @Column(name = "IMAGE_URI")
  public String imageUri;

  @Column(name = "ACTIVE")
  public Boolean active;

  @Column(name = "CREATED_AT")
  public Date createdAt;

  @Column(name = "UPDATED_AT")
  public Date updatedAt;

  public Product() {
  }

  public Product(ProductDto dto) {
    name = dto.getName();
    price = dto.getPrice();
    discount = dto.getDiscount();
    description = dto.getDescription();
    imageUri = dto.getImageUri();
    active = dto.getActive();
  }

  public Integer getId() {
    return this.id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getName() {
    return this.name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public Double getPrice() {
    return this.price;
  }

  public void setPrice(Double price) {
    this.price = price;
  }

  public Double getDiscount() {
    return this.discount;
  }

  public void setDiscount(Double discount) {
    this.discount = discount;
  }

  public String getDescription() {
    return this.description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getImageUri() {
    return this.imageUri;
  }

  public void setImageUri(String imageUri) {
    this.imageUri = imageUri;
  }

  public Boolean getActive() {
    return this.active;
  }

  public void setActive(Boolean active) {
    this.active = active;
  }

}
