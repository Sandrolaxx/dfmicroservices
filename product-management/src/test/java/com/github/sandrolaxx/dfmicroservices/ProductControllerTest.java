package com.github.sandrolaxx.dfmicroservices;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;

import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import javax.ws.rs.core.Response.Status;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.sandrolaxx.dfmicroservices.dto.ProductCreateDto;
import com.github.sandrolaxx.dfmicroservices.entities.Product;
import com.github.sandrolaxx.dfmicroservices.util.TokenUtils;

@DBRider
//Tabela no postgres é minusculo, por default na criação 
//da base de testes o DBRider deixa maiusculo, correção abaixo
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(ProductTestLifecycleManager.class)
public class ProductControllerTest {

    private String token;

    @BeforeEach
    public void genereteToken() throws Exception {
        token = TokenUtils.generateTokenString("/JWTDonaFrostClaims.json", null);
    }

    private RequestSpecification givenNoToken() {
        return RestAssured.given()
        .contentType(ContentType.JSON);
    }

    private RequestSpecification given() {
        return RestAssured.given()
        .contentType(ContentType.JSON).header(new Header("Authorization", "Bearer " + token));
    }

    @Test
    @DataSet("products-cenario1.json")
    public void testFindProducts() {
        String result = given()
            .when().get("/dona-frost/v1/product")
            .then()
            .statusCode(200)
            .extract().asString();

            Approvals.verifyJson(result);
    }

    @Test
    @DataSet("products-cenario1.json")
    public void testFindUsersNoToken() {
        givenNoToken()
            .when().get("/dona-frost/v1/product")
            .then()
            .statusCode(401)
            .extract().asString();
    }

    @Test
    @DataSet(value = "products-cenario1.json")
    public void testCreateProduct() {
        ProductCreateDto dto = new ProductCreateDto();

        dto.setName("Lasanha Italiana");
        dto.setPrice(22.90);
        dto.setDescription("A descripton for the test");
        dto.setImageUri("A new uri");
        dto.setActive(true);

        String result = given()
            .body(dto)
            .when().post("/dona-frost/v1/product")
            .then()
            .statusCode(201)
            .extract().asString();

        //Somente aplicação rodando, não é possível criar transação no test    
        Approvals.verifyJson(result);
    }

    @Test
    @DataSet("products-cenario1.json")
    public void testChangeProduct() {
        ProductCreateDto dto = new ProductCreateDto();

        dto.setName("Lasanha Italiana");
        dto.setPrice(22.90);
        dto.setDescription("A descripton for the test");
        dto.setImageUri("A new uri");
        dto.setActive(true);

        Integer idProductToChange = 1;

        given()
            .with().pathParam("id", idProductToChange)
            .body(dto)
            .when().put("/dona-frost/v1/product/{id}")
            .then()
            .statusCode(Status.NO_CONTENT.getStatusCode())
            .extract().asString();

        Product findById = Product.findById(idProductToChange);
        
        Assert.assertEquals(dto.getName(), findById.getName());
        Assert.assertEquals(dto.getDescription(),findById.getDescription());
        Assert.assertEquals(dto.getPrice(), findById.getPrice());
    }

    @Test
    @DataSet("products-cenario1.json")
    public void testDeleteProduct() {

        Integer idProductToDelete = 1;

        given()
            .with().pathParam("id", idProductToDelete)
            .when().delete("/dona-frost/v1/product/{id}")
            .then()
            .statusCode(Status.NO_CONTENT.getStatusCode())
            .extract().asString();

            Product findById = Product.findById(idProductToDelete);
        
        Assert.assertEquals(findById, null);

    }

}