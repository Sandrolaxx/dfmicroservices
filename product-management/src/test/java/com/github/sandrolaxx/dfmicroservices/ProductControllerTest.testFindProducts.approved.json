[
  {
    "active": true,
    "createdAt": "03/04/2021 13:30:42",
    "description": "Queijo, molho, carne… Sim, estou falando daquela delícia que é o bife à parmegiana! Dá até água na boca, né?",
    "id": 1,
    "imageUri": "sim.githb.com",
    "name": "Bife à parmegiana",
    "price": 22.2,
    "updatedAt": "03/04/2021 13:30:42"
  },
  {
    "active": true,
    "createdAt": "03/04/2021 14:48:42",
    "description": "Feita com camadas alternadas de massa, queijo e molho de carne.",
    "id": 2,
    "imageUri": "testedoteste.com/imagem",
    "name": "Lasanha Italiana",
    "price": 13.32,
    "updatedAt": "03/04/2021 14:48:42"
  },
  {
    "active": true,
    "createdAt": "03/04/2021 12:58:42",
    "description": "Alface americana, salsinha, tomate cereja, queijo minas. Acompanha: Croutons.",
    "id": 4,
    "imageUri": "testedoteste.com/imagem 3",
    "name": "Salada da Chef",
    "price": 53.0,
    "updatedAt": "03/04/2021 12:58:42"
  }
]